src/ClientData/include/ClientData/CaptureData.h

```diff
@@ -276,9 +276,9 @@ class CaptureData {
   [[nodiscard]] std::optional<ThreadStateSliceInfo> FindThreadStateSliceInfoFromTimestamp(
       int64_t thread_id, uint64_t timestamp) const;
 
-  [[nodiscard]] std::unique_ptr<ScopeStatsCollection> CreateScopeStatsCollection(
+  [[nodiscard]] std::shared_ptr<const ScopeStatsCollection> CreateScopeStatsCollection(
```

Comment:
Even though we have agreed `ScopeStatsCollection` should be owned via `shared_ptr`, the factory function doesn't have to assume shared ownership. So I suggest to return `unique_ptr` here and convert it to `shared_ptr` when needed.